src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BazelModuleResolutionFunction.java

```diff
@@ -259,14 +266,41 @@ private static ImmutableMap<ModuleKey, Module> computeFinalDepGraph(
       ImmutableMap<String, ModuleOverride> overrides,
       ExtendedEventHandler eventHandler)
       throws BazelModuleResolutionFunctionException, InterruptedException {
-    ImmutableMap.Builder<ModuleKey, Module> finalDepGraph = ImmutableMap.builder();
-    for (Map.Entry<ModuleKey, InterimModule> entry : resolvedDepGraph.entrySet()) {
-      finalDepGraph.put(
-          entry.getKey(),
-          moduleFromInterimModule(
-              entry.getValue(), overrides.get(entry.getKey().getName()), eventHandler));
+    ExecutorService executorService = Executors.newWorkStealingPool(8);
+    var future =
+        executorService.submit(
+            () ->
+                resolvedDepGraph.entrySet().stream()
+                    .parallel()
+                    .map(
```

Comment:
since `Module` both has a `key()` method, this could be simplified to:

```java
resolvedDepGraph.values().parallelStream()
    .map(...)  // calls moduleFromInterimModule
    .collect(toImmutableMap(Module::getKey, m -> m));
```